src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/shmem/shmem_port.cc

```diff
@@ -73,7 +74,13 @@ void ShmemRecvPort::QueueRecv() {
       ret = shm_->Load([this](void* data){
         MetaDataPtr metadata_res = std::make_shared<MetaData>();
         std::memcpy(metadata_res.get(), data, sizeof(MetaData));
-        metadata_res->mdata = malloc(this->nbytes_ - sizeof(MetaData));
+        int mem_size = (this->nbytes_ - sizeof(MetaData) + 7) & (~0x7);
```

Comment:
- define a func that convert void* to MetaDataPtr
- use it at `ShmemRecvPort::QueueRecv`, `ShmemBlockRecvPort::Recv` and `ShmemBlockRecvPort::Peek`